spec/cyanobyte.schema.json

```diff
@@ -0,0 +1,216 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://github.com/cyanobyte/cyanobyte.schema.json",
+  "title": "CyanoByte Specification Validator",
+  "description": "This file is used to validate whether a given document meets the CyanoByte specification.",
+  "type": "object",
+  "properties": {
+    "cyanobyte": {
+      "description": "The version of the CyanoByte specification this file follows.",
+      "type": "string"
+    },
+    "info": {
+      "description": "Info block of the file",
+      "type": "object",
+      "properties": {
+        "title": {
+          "description": "The title of the document. This should be used as the class name in the auto-generated code and therefore should be short (e.g. the part name).",
+          "type": "string"
+        },
+        "description": {
+          "description": "A description of the document.",
+          "type": "string"
+        },
+        "contact": {
+          "description": "Contact info for the maintainer.",
+          "type": "object",
+          "properties": {
+            "name": {
+              "description": "Name of the maintainer.",
+              "type": "string"
+            },
+            "url": {
+              "description": "URL of the maintainer organization.",
+              "type": "string",
+              "format": "uri"
+            },
+            "email": {
+              "description": "Email address for the maintainer.",
+              "type": "string",
+              "format": "email"
+            }
+          },
+          "required": [
+            "name",
+            "url",
+            "email"
+          ]
+        },
+        "license": {
+          "description": "License this document uses. TODO: Add validation other than it being a string.",
```

Comment:
Shouldn't this field break down into a name and optional url?

Reply:
Currently this doesn't validate with the example JSON. Recommend

```json
"license": {
          "description": "License this document uses. TODO: Add validation other than it being a string.",
          "type": "object",
          "properties": {
             "name": {
                "description": "A SPDX compliant license identifier",
                "type": "string"
             },
             "url": {
                "description": "If the license name is not SPDX compliant, this can link to the license",
                "type": "string"
             }
          }
        },
```

Reply:
Did we want to allow a url? I thought the conclusion from #6 was to omit it. Fine with optional if we want to do that.

Reply:
To be pedantic, having an optional URL would fit the 0.01% of cases where there is a different, non-standard license.

Reply:
Updated.